modules/database/test/std/rec/seqTest.c

```diff
@@ -0,0 +1,55 @@
+/*************************************************************************\
+* Copyright (c) 2020 Gabriel Fedel 
+* EPICS BASE is distributed subject to a Software License Agreement found
+* in file LICENSE that is included with this distribution.
+\*************************************************************************/
+
+#include "dbUnitTest.h"
+#include "testMain.h"
+#include "errlog.h"
+#include "dbAccess.h"
+#include <epicsThread.h>
+
+void recTestIoc_registerRecordDeviceDriver(struct dbBase *);
+
+/*
+ * This test verifies the behavior of seq using Specified for SELM
+ * The behavior should be the same for all the DOLx
+ * */
+static
+void testSeqSpecified(void){
+    int i;
+    for (i=0; i < 16; i++) {
+        testdbPutFieldOk("seq0.SELN", DBR_USHORT, i);
+
+        testdbPutFieldOk("ai0", DBR_LONG, 0);
+
+        testdbPutFieldOk("seq0.PROC", DBR_USHORT, 1);
+
+        epicsThreadSleep(0.1);
```

Comment:
Is this sleep required?  It might not be.  If it is, then it will result in false positive failures during CI builds.

Reply:
I've tried without this and it didn't work. My impression is it is needed to wait some time to ai record could be processed.  

Do you now some way to avoid this?

Reply:
I think `testSyncCallback()` from dbUnitTest.h is the right tool.  The issue is that seqRecord is always asynchronous, even when `DLY*` is zero.  (cf. `processNextLink()`) So `dbProcess()` returns before the callbacks have run.  In this case (`SELM==Specified`) I think there will be only one `callbackRequest()`, so only one `testSyncCallback()` will be needed.

Reply:
Thank you. I was looking for testSyncCallback before I saw your comment. I've just made a new commit using it.